doc/sonic-sub-port-intf-hld.md

```diff
@@ -0,0 +1,475 @@
+    * [2.1 Configuration](#21-configuration)
+        * [2.1.1 config_db.json](#211-config-db-json)
+        * [2.1.2 CONFIG_DB](#212-config-db)
+        * [2.1.3 CONFIG_DB schemas](#213-config-db-schemas)
+    * [2.2 APPL_DB](#22-appl-db)
+    * [2.3 STATE_DB](#23-state-db)
+    * [2.4 SAI](#24-sai)
+        * [2.4.1 Create a sub port interface](#241-create-a-sub-port-interface)
+        * [2.4.2 Set a sub port interface mtu](#242-set-a-sub-port-interface-mtu)
+        * [2.4.3 Remove a sub port interface](#243-remove-a-sub-port-interface)
+    * [2.5 Linux integration](#25-linux-integration)
+        * [2.5.1 Host sub port interfaces](#251-host-sub-port-interfaces)
+        * [2.5.2 Route, neighbor subsystems](#252-route-neighbor-subsystems)
+  * [3 Event flow diagrams](#3-event-flow-diagrams)
+    * [3.1 Sub port interface creation](#31-sub-port-interface-creation)
+    * [3.2 Sub port interface runtime mtu change](#32-sub-port-interface-runtime-mtu-change)
+    * [3.3 Sub port interface removal](#33-sub-port-interface-removal)
+  * [4 CLI](#4-cli)
+    * [4.1 Config commands](#41-config-commands)
+    * [4.2 Show commands](#42-show-commands)
+  * [5 Warm reboot support](#5-warm-reboot-support)
+  * [6 Unit test](#6-unit-test)
+    * [6.1 Sub port interface creation](#61-sub-port-interface-creation)
+        * [6.1.1 Create a sub port interface](#611-create-a-sub-port-interface)
+        * [6.1.2 Add an IP address to a sub port interface](#612-add-an-ip-address-to-a-sub-port-interface)
+    * [6.2 Sub port interface mtu change](#62-sub-port-interface-mtu-change)
+    * [6.3 Sub port interface removal](#63-sub-port-interface-removal)
+        * [6.3.1 Remove an IP address from a sub port interface](#631-remove-an-ip-address-from-a-sub-port-interface)
+        * [6.3.2 Remove all IP addresses from a sub port interface](#632-remove-all-ip-addresses-from-a-sub-port-interface)
+        * [6.3.3 Remove a sub port interface](#633-remove-a-sub-port-interface)
+  * [7 Scalability](#7-scalability)
+  * [8 Port channel renaming](#8-port-channel-renaming)
+
+<!-- /TOC -->
+
+# Revision history
+| Rev |    Date     |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 07/01/2019  | Wenda Ni           | Initial version                   |
+
+# Scope
+A sub port interface is a logical interface that can be created on a physical port or a port channel.
+A sub port interface serves as an interface to either a .1D bridge or a VRF, but not both.
+This design focuses on the use case of creating a sub port interface on a physical port or a port channel and using it as a router interface to a VRF.
+
+![](https://github.com/wendani/SONiC/blob/sub_port_master/images/sub_interface_hld/sub_intf_rif.png)
+
+Multiple L3 sub port interfaces, each characterized by a VLAN id in the 802.1q tag, can be created on a physical port or a port channel.
```

Comment:
previous `.1D`, now `802.1q`. Could you clarify? This line is difficult to understand.

Reply:
The tag is called 802.1q tag. Just a name to refer to the 32-bit fields in the Ethernet frame.